src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -311,7 +311,7 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
                                 continue;
                             }
 
-                            _env.ValidatePageChecksum(modifiedPage, ptr);
+                            _env.ValidateInMemoryPageChecksum(modifiedPage, ptr);
```

Comment:
Instead of introducing a new method can we do:

```
var checksum = StorageEnvironment.CalculatePageChecksum((byte*)pageHeader, pageNumber, out var expectedChecksum);
if (checksum != expectedChecksum) 
    ThrowInvalidChecksumOnPageFromJournal(pageNumber, current, expectedChecksum, checksum, pageHeader);
```
? 

We already do it this way in few other places

Reply:
ping

Reply:
I would want to avoid that for the following reason, when I introduce a new hash function I want all that to be encapsulated inside. The same would happen when I work on removing all the IFs that are needed alongside Voron because we have encrypted databases and normal ones. There is work to be done there to do the right thing, but avoiding leaking this type of internal knowledge is the first step into that direction.

Reply:
Let's use then your `ValidateInMemoryPageChecksum()` in other places where we force the validation

Reply:
That's the only call where that applies for now. In fact had to write it specifically for that case.

Reply:
What I meant is that we have 2 other places where we do the same thing:

https://github.com/ravendb/ravendb/blob/e616b8988b7a514113214e2f0bbd9269e04a911b/src/Voron/Impl/Journal/JournalReader.cs#L177-L179

https://github.com/ravendb/ravendb/blob/e616b8988b7a514113214e2f0bbd9269e04a911b/src/Voron/Impl/Journal/WriteAheadJournal.cs#L1492-L1494

Can we call there your method `ValidateInMemoryPageChecksum()`?

Reply:
No, we lose knowledge in the exception information if we do so. I would avoid that.

Reply:
OKay